domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -376,10 +422,27 @@ void setClient(PgPool client) {
   }
 
   /**
-   * Close the SQL client of this PostgresClient instance.
+   * Close the SQL writer and reader clients of this PostgresClient instance in succession.
    * This is idempotent: additional close invocations are always successful.
    */
-  public Future<Void> closeClient() {
+  public Future<Void> closeReadClient() {
+    if (readClient == null) {
+      return Future.succeededFuture();
+    }
+    PgPool clientToClose = readClient;
+    readClient = null;
+    CONNECTION_POOL.removeMultiKey(vertx, tenantId);  // remove (vertx, tenantId, this) entry
+    if (sharedPgPool) {
+      return Future.succeededFuture();
+    }
+    return clientToClose.close();
+  }
+
+  /**
+   * Close the SQL writer and reader clients of this PostgresClient instance in succession.
+   * This is idempotent: additional close invocations are always successful.
+   */
+  public Future<Void> closeWriteClient() {
```

Comment:
closeWriteClient is unsed. We should not add useless code. So I suggest that to be removed.